### PR TITLE
Rework passing array literal to native commands

### PR DESCRIPTION
**Pull request by lzybkr**
Now that the Ast is passed through to the NativeCommandParameterBinder, there is a cleaner fix to ensuring we pass the correct text to a command that uses PowerShell array syntax.

### CODE REVIEW

File: test/powershell/Language/Scripting/NativeExecution/NativeCommandArguments.Tests.ps1 [hunk 1]
```diff
@@ -43,4 +43,21 @@ Describe "Native Command Arguments" -tags "CI" {
             $lines[$i] | Should Be "Arg $i is <$($expected[$i])>"
         }
     }
+
+    It "Should handle PowerShell arrays with or without spaces correctly: <arguments>" -TestCases @(
+        @{arguments = "a 1,2"; expected = "a", "1,2"}
```
Comment (vors) [inline test/powershell/Language/Scripting/NativeExecution/NativeCommandArguments.Tests.ps1]: let's add something boring, like `a,1,2` here as well

Comment (lzybkr) [reply test/powershell/Language/Scripting/NativeExecution/NativeCommandArguments.Tests.ps1]: There is no difference between 2 and 3 elements - it's just spaces around the argument(s) that matter.

File: src/System.Management.Automation/engine/NativeCommandParameterBinder.cs [hunk 2]
```diff
@@ -158,14 +157,19 @@ internal String Arguments
...
         {
             IEnumerator list = LanguagePrimitives.GetEnumerator(obj);
-            bool needSeparator = false;
 
+            Diagnostics.Assert(argArrayAst == null
+                || obj is object[] && ((object[])obj).Length == argArrayAst.Elements.Count,
+                "array argument and ArrayLiteralAst differ in number of elements");
+
+            int currentElement = -1;
+            string separator = "";
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/NativeCommandParameterBinder.cs]: Should we use `String.Empty`?

File: src/System.Management.Automation/engine/NativeCommandParameterBinder.cs [hunk 3]
```diff
@@ -349,6 +352,35 @@ internal static bool NeedQuotes(string stringToCheck)
...
+
+            var next = arrayLiteralAst.Elements[index];
+            var prev = arrayLiteralAst.Elements[index - 1];
+
+            var arrayExtent = arrayLiteralAst.Extent;
+            var afterPrev = prev.Extent.EndOffset;
+            var beforeNext = next.Extent.StartOffset - 1;
+
+            if (afterPrev == beforeNext)
+                return ",";
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/NativeCommandParameterBinder.cs]: Could you please use one line formatting as above in line 357?
Below too in 376 and 379.

File: test/powershell/Language/Scripting/NativeExecution/NativeCommandArguments.Tests.ps1 (hunk 1 above)
Comment (iSazonov) [reply test/powershell/Language/Scripting/NativeExecution/NativeCommandArguments.Tests.ps1]: Should we add `1,2 3`?

**Review by iSazonov — COMMENTED** [review]
Leave a comment

File: test/powershell/Language/Scripting/NativeExecution/NativeCommandArguments.Tests.ps1 (hunk 1 above)
Comment (lzybkr) [reply test/powershell/Language/Scripting/NativeExecution/NativeCommandArguments.Tests.ps1]: I really didn't need `a 1,2`, `1,2` would have been sufficient, but I had a bug in the test with array vs. scalar and the first thought I had was to add `a ` to force array. I'm fixing by using `@()` instead to force array.

File: src/System.Management.Automation/engine/NativeCommandParameterBinder.cs (hunk 2 above)
Comment (lzybkr) [reply src/System.Management.Automation/engine/NativeCommandParameterBinder.cs]: No reason to imo.

**Review by daxian-dbw — APPROVED** [review]
LGTM.

**Pull request merged**